concurrent/src/test/java/io/airlift/concurrent/TestBoundedThreadPool.java

```diff
@@ -0,0 +1,104 @@
+package io.airlift.concurrent;
+
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.Test;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static com.google.common.util.concurrent.Uninterruptibles.awaitUninterruptibly;
+import static io.airlift.concurrent.BoundedThreadPool.newBoundedThreadPool;
+import static io.airlift.concurrent.Threads.threadsNamed;
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+
+public class TestBoundedThreadPool
+{
+    private ExecutorService executor;
+
+    @AfterClass(alwaysRun = true)
+    public void tearDown()
+            throws Exception
+    {
+        executor.shutdownNow();
+    }
+
+    @Test
+    public void testCounter()
```

Comment:
I copied all these tests from TestBoundedExecutor and am not comfortable changing them.